File: LING570/code-samples/c++/ncat.cpp
#include <iostream>
#include <string>

using namespace std;

// This program takes a single integer arguments and prints that many lines
// from STDIN to STDOUT.

int main(int argc, char* argv[]) {
	int n;
	string line;
	
	// If no arguments are specified, n=-1 will cause the program to loop until
	// STDIN is exhausted.
	n = argc >= 2 ? atoi(argv[1]) : -1;
	while(n-- && getline(cin, line)) {
		cout << line << endl;
	};
	return 0;
}

File: LING570/code-samples/java/NCat.java
import java.io.*;


class NCat {
public static void main (String[] args) throws IOException {
	String line;
    BufferedReader stdin = new BufferedReader
      (new InputStreamReader(System.in));

	// If there are no command line arguments, n=-1 will cause the loop to
	// continue until STDIN is exhausted.
	int n = args.length == 0 ? -1 : Integer.parseInt(args[0]);

	while (true) {
		if (n-- == 0)
			break;
		line = stdin.readLine();
		if (line == null)
			break;
		System.out.println(line);
	}
  }
}

